### PR TITLE
bv_pointert cleanup

### PR DESCRIPTION
**Pull request by kroening**
- [X] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/solvers/flattening/bv_pointers.cpp [hunk 1]
```diff
@@ -68,58 +68,33 @@ void bv_endianness_mapt::build_big_endian(const typet &src)
 endianness_mapt
 bv_pointerst::endianness_map(const typet &type, bool little_endian) const
 {
-  return bv_endianness_mapt{type, little_endian, ns, bv_pointers_width};
+  return bv_endianness_mapt{type, little_endian, ns, bv_width};
```
Comment (tautschnig) [inline src/solvers/flattening/bv_pointers.cpp:71]: That second commit is a part-revert of 5cf2b07c12f. If we go for removing the capability that the commit was trying to add we should ponder a clean revert instead.

**Review by tautschnig — COMMENTED** [review]
I'm happy with the first and third commit, but the middle one part reverts work that I previously did. Having a clean revert would be more helpful.

File: src/solvers/flattening/bv_pointers.cpp (hunk 1 above)
Comment (kroening) [reply src/solvers/flattening/bv_pointers.cpp:71]: We absolutely want to keep the parts of that commit that pass the pointer type around!

**Review by tautschnig — APPROVED** [review]
Approving with the caveat that I may propose to revert the middle commit at a future date.

**Pull request merged**